lib/buildingsync/model_articulation/spatial_element.rb

```diff
@@ -281,18 +281,18 @@ def add_element_in_xml_file(building_element, ns, field_name, field_value)
     end
 
     def write_parameters_to_xml_for_spatial_element(ns, xml_element)
-      add_element_in_xml_file(xml_element, ns, 'TotalFloorArea', @total_floor_area)
-      add_element_in_xml_file(xml_element, ns, 'BuildingType', @bldg_type)
-      add_element_in_xml_file(xml_element, ns, 'SystemType', @system_type)
-      add_element_in_xml_file(xml_element, ns, 'BarDivisionMethod', @bar_division_method)
-      add_element_in_xml_file(xml_element, ns, 'FractionArea', @fraction_area)
-      add_element_in_xml_file(xml_element, ns, 'SpaceTypesFloorArea', @space_types_floor_area)
-      add_element_in_xml_file(xml_element, ns, 'ConditionedFloorAreaHeatedOnly', @conditioned_floor_area_heated_only)
-      add_element_in_xml_file(xml_element, ns, 'ConditionedFloorAreaCooledOnly', @conditioned_floor_area_cooled_only)
-      add_element_in_xml_file(xml_element, ns, 'ConditionedFloorAreaHeatedCooled', @conditioned_floor_area_heated_cooled)
-      add_element_in_xml_file(xml_element, ns, 'ConditionedBelowGradeFloorArea', @conditioned_below_grade_floor_area)
-      add_element_in_xml_file(xml_element, ns, 'CustomConditionedAboveGradeFloorArea', @custom_conditioned_above_grade_floor_area)
-      add_element_in_xml_file(xml_element, ns, 'CustomConditionedBelowGradeFloorArea', @custom_conditioned_below_grade_floor_area)
+      add_user_defined_field_to_xml_file(xml_element, ns, 'TotalFloorArea', @total_floor_area)
+      add_user_defined_field_to_xml_file(xml_element, ns, 'BuildingType', @bldg_type)
+      add_user_defined_field_to_xml_file(xml_element, ns, 'SystemType', @system_type)
+      add_user_defined_field_to_xml_file(xml_element, ns, 'BarDivisionMethod', @bar_division_method)
+      add_user_defined_field_to_xml_file(xml_element, ns, 'FractionArea', @fraction_area)
+      add_user_defined_field_to_xml_file(xml_element, ns, 'SpaceTypesFloorArea', @space_types_floor_area)
```

Comment:
What does `SpaceTypesFloorArea` and `FractionArea` refer to?

Reply:
@corymosiman12 
fraction_area defined the fraction of the section/building/element of the total area
space_types_floor_area is actually a hash that contains the floor area of each space type